excellent/types/base.go

```diff
@@ -0,0 +1,262 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/nyaruka/goflow/utils"
+
+	"github.com/shopspring/decimal"
+)
+
+func init() {
+	decimal.MarshalJSONWithoutQuotes = true
+}
+
+// XType represents a type in Excellent
+type XType int
+
+// the supported types
+const (
+	XTypeString XType = iota
+	XTypeNumber
+	XTypeBool
+	XTypeTime
+	XTypeArray
+	XTypeObject
+	XTypeError
+	XTypeNil
```

Comment:
Do we need XType anymore? This was mostly added for equality and we are really only supporting equality in excellent for bool and numeric types, so that's vastly simplified. Seems we could probably remove these and just use normal type assertions instead.